rclc_examples/CMakeLists.txt

```diff
@@ -57,9 +57,11 @@ add_executable(example_pingpong src/example_pingpong.cpp)
 ament_target_dependencies(example_pingpong rcl rclc std_msgs)
 
 add_executable(example_action_server src/example_action_server.c)
+target_link_libraries(example_action_server ${CMAKE_THREAD_LIBS_INIT})
```

Comment:
Something needs to call `find_package(Threads)` - that's the module that provides this variable: https://cmake.org/cmake/help/latest/module/FindThreads.html

Also, the `Threads::Threads` target is the thing to link against.

```cmake
target_link_libraries(example_action_server Threads::Threads)
```

Reply:
`find_package` lines added here: https://github.com/ros2/rclc/pull/237

And for the `Threads::Threads`, it looks like `CMAKE_THREAD_LIBS_INIT` should handle the link target.

Reply:
> And for the Threads::Threads, it looks like CMAKE_THREAD_LIBS_INIT should handle the link target.

[`Threads::Threads` provides that](https://github.com/Kitware/CMake/blob/410dd6cf611f31ebca74e1db4dc439e6fcaf34b4/Modules/FindThreads.cmake#L252-L254).

The advantage of linking to a modern CMake target instead of a result variables is the target usually has logic for different environments. For example, linking to `Threads::Threads` will use the [`-pthread` compile option](https://stackoverflow.com/questions/23250863/difference-between-pthread-and-lpthread-while-compiling)  when available.

https://github.com/Kitware/CMake/blob/410dd6cf611f31ebca74e1db4dc439e6fcaf34b4/Modules/FindThreads.cmake#L246-L250

Reply:
resolved in https://github.com/ros2/rclc/pull/237/commits/da1cebeea58447ff216323994f11e649bdd1e3a4